src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/ComponentAvailabilityService.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Tooltip;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Hover;
```

Comment:
nit: Should this be in Hover?

Reply:
Sorry, I didn't follow. It's in the `Hover` folder and is in the same namespace as other classes in this folder. Is there someplace else it should move to?

Reply:
I mean this service doesn't seem specific to hover/tooltip. Is there a better location for the definition of this service?

Reply:
This service is derived from implementation details of hover/tooltips, which need to show which projects a component is available in for multi-targeting scenarios. To date, no other features need this, so I'm comfortable leaving it alongside hover/tooltip as an implementation detail.